File: Selenium/Activities/Activity20.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.select import Select
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as FirefoxService

serv = FirefoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=serv) as driver:
    driver.get("https://training-support.net/selenium/javascript-alerts")
    wait = WebDriverWait(driver, 10)
    driver.find_element(By.ID, "prompt").click()
    wait.until(expected_conditions.alert_is_present())
    prompt_alert = driver.switch_to.alert
    print("Text: ", prompt_alert.text)
    prompt_alert.send_keys("Awesome!")
    prompt_alert.accept()
    driver.quit()
    

File: Selenium/Activities/Activity20.java
package Activities;
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.Alert;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedCondition;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;
import java.util.List;

public class Activity20 {
    public static void main(String[] args) throws InterruptedException {
        WebDriverManager.firefoxdriver().setup();
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriver newDriver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(newDriver, Duration.ofSeconds(10));
        newDriver.get("https://training-support.net/selenium/javascript-alerts");
        System.out.println("Title of page: " + newDriver.getTitle());
        newDriver.findElement(By.id("prompt")).click();
        Thread.sleep(300);
        Alert alert1 = wait.until(ExpectedConditions.alertIsPresent());
        newDriver.switchTo().alert();
        System.out.println("Text is: " + alert1.getText());
        alert1.sendKeys("Awesome!");
        Thread.sleep(300);
        alert1.accept();
        Thread.sleep(300);
        newDriver.quit();
    }
}